File: frontend/src/app/types/limit-adjustment.ts
export type LimitAdjustmentStatus = 'pending' | 'approved' | 'rejected';

export interface TemporaryLimitAdjustment {
  id: number;
  worker_id: number;
  worker_code: string;
  worker_name: string;
  status: LimitAdjustmentStatus;
  effective_date: string;
  expiry_date: string;
  adjusted_limit_msv: number;
  reason: string;
  created_by: number;
  created_at: string;
  reviewed_by?: number;
  reviewed_at?: string;
  rejection_reason: string;
}

export interface LimitAdjustmentInput {
  worker_id: number;
  effective_date: string;
  expiry_date: string;
  adjusted_limit_msv: number;
  reason: string;
}

File: backend/internal/dto/limit_adjustment.go
package dto

import "time"

type CreateLimitAdjustmentRequest struct {
	EffectiveFrom    time.Time `json:"effective_from" validate:"required"`
	EffectiveTo      time.Time `json:"effective_to" validate:"required,gtfield=EffectiveFrom"`
	AdjustedLimitMSV float64   `json:"adjusted_limit_msv" validate:"required,gt=0,lte=1000"`
	Reason           string    `json:"reason" validate:"required,min=3,max=500"`
}

type ReviewLimitAdjustmentRequest struct {
	Decision string `json:"decision" validate:"required,oneof=approve reject"`
	Note     string `json:"note" validate:"max=500"`
}

type LimitAdjustmentResponse struct {
	ID               uint       `json:"id"`
	WorkerID         uint       `json:"worker_id"`
	WorkerCode       string     `json:"worker_code"`
	WorkerName       string     `json:"worker_name"`
	EffectiveFrom    time.Time  `json:"effective_from"`
	EffectiveTo      time.Time  `json:"effective_to"`
	AdjustedLimitMSV float64    `json:"adjusted_limit_msv"`
	Reason           string     `json:"reason"`
	Status           string     `json:"status"`
	ReviewedBy       *uint      `json:"reviewed_by,omitempty"`
	ReviewedAt       *time.Time `json:"reviewed_at,omitempty"`
	ReviewNote       string     `json:"review_note"`
	CreatedBy        uint       `json:"created_by"`
	CreatedAt        time.Time  `json:"created_at"`
}
